### PR TITLE
Archetype cleanup

### PR DESCRIPTION
**Pull request by k-r-g**
Make the archetype a more useful example of getting started with server-side rendering.

Also fixes some stuff I messed up in the archetype with the previous database migrations work. (Sorry.)

### CODE REVIEW

File: gemini-resin-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
+    <dependency>
+      <groupId>com.techempower</groupId>
+      <artifactId>gemini-jdbc</artifactId>
+      <version>${geminiVersion}</version>
+    </dependency>
+    <!-- gemini-hikaricp is an alternative to gemini-jdbc -->
+    <!-- <dependency> -->
+    <!-- <groupId>com.techempower</groupId> -->
+    <!-- <artifactId>gemini-hikaricp</artifactId> -->
+    <!-- </dependency> -->
```
Comment (msmith-techempower) [inline gemini-resin-archetype/src/main/resources/archetype-resources/pom.xml]: I think this could be the other way around - if my instinct is "my new Gemini projects will always use hikaricp," then I think we should support that recommendation by default.

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/java/com/techempower/handlers/HomeHandler.java [hunk 2]
```diff
@@ -4,19 +4,34 @@
...
+  public boolean example()
+  {
+    return mustache("example");
+  }
   
+  @PathDefault
+  public boolean notFound()
+  {
+    //mustache("common/error");
+    return notFound("Not found");
```
Comment (msmith-techempower) [inline gemini-resin-archetype/src/main/resources/archetype-resources/src/main/java/com/techempower/handlers/HomeHandler.java]: Is the use of `MethodSegmentHandler` simply for the `notFound` method?

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/App-Dev.conf [hunk 3]
```diff
@@ -16,6 +16,7 @@ Extends = App-Base.conf
 DeploymentDescription = Development/${Servlet.MachineName}
 
 # Database connectivity for Development.
+# TODO: EDIT THIS
```
Comment (msmith-techempower) [inline gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/App-Dev.conf]: The meta here is hard to follow - is this a `TODO` for you on this PR, or for the developer who builds a Gemini project from this archetype?

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/hikari.properties [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# This is only used if Application.constructConnectorFactory() returns a
+# HikariCPConnectorFactory. Otherwise see the Gemini configuration files
+# for database connection settings.
+dataSourceClassName=org.postgresql.ds.PGSimpleDataSource
+dataSource.user=${databaseUsername}
+dataSource.password=${databasePassword}
+dataSource.databaseName=${databaseName}
+dataSource.portNumber=${databasePort}
+dataSource.serverName=${databaseHost}
```
Comment (msmith-techempower) [inline gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/hikari.properties]: This is one of those things that ends up confusing down the line. This states that if I configure my application, which may have database credentials in several gemini-specific configuration files, to use HikariCP then the expectation is that this `hiarki.properties` file will be used while the gemini-specific ones will be ignored as it pertains to database configuration.

So, one could easily find himself in the situation where a project switches over and the production (as an example) config says one thing but the hikari says another, and a new dev would be very confused.

I don't have a perfect solution to this, but it seems likely to be confusing unless we do [something] to make it less so.

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/resin.xml [hunk 5]
```diff
@@ -35,7 +35,7 @@
       <filter-mapping url-pattern='*' filter-name="gzip" />
     </web-app-default>
 
-    <resin:import path="${RESIN_HOME}/conf/app-default.xml" />
+    <resin:import path="${resin.home}/conf/app-default.xml" />
```
Comment (msmith-techempower) [inline gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/resin.xml]: This may no longer be true, but I believe that I had `RESIN_HOME` here because 1. Eclipse did not support passing through environment variables to the java execution and 2. did not support dot-notated vars in the configuration dialog. Ignore if this just works in Eclipse now.

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml [hunk 6]
```diff
@@ -45,6 +45,7 @@
   <servlet>
     <servlet-name>Server</servlet-name>
     <servlet-class>com.techempower.Server</servlet-class>
+    <load-on-startup/>
```
Comment (msmith-techempower) [inline gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml]: I have never included this but it also seemed like this was true as a default... am I crazy?

File: gemini/src/main/java/com/techempower/gemini/pyxis/BasicSecurity.java [hunk 7]
```diff
@@ -30,9 +30,11 @@
 import java.util.*;
 import java.util.concurrent.*;
 
+import com.sun.mail.handlers.*;
```
Comment (msmith-techempower) [inline gemini/src/main/java/com/techempower/gemini/pyxis/BasicSecurity.java]: This import can't be right.

File: gemini-resin-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (k-r-g) [reply gemini-resin-archetype/src/main/resources/archetype-resources/pom.xml]: I don't disagree, however the HikariCP integration hasn't received any real-world usage yet outside of me doing TFB tests. Happy to switch it, but I didn't want to presume.

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/App-Dev.conf (hunk 3 above)
Comment (k-r-g) [reply gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/App-Dev.conf]: The Gemini configuration files already had a number of "TODO" comments intended for the developer building a Gemini project from the archetype. I just added some more.

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml (hunk 6 above)
Comment (k-r-g) [reply gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml]: Happy to discard this. My preference is for the app to start on its own so I don't have to wait when I make my first request.

Documentation: https://stackoverflow.com/a/7817279

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/hikari.properties (hunk 4 above)
Comment (k-r-g) [reply gemini-resin-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/configuration/hikari.properties]: Good point. Recommend discarding this unless we make gemini-hikaricp the default in our other conversation.

File: gemini/src/main/java/com/techempower/gemini/pyxis/BasicSecurity.java (hunk 7 above)
Comment (k-r-g) [reply gemini/src/main/java/com/techempower/gemini/pyxis/BasicSecurity.java]: Oops, sorry. Pushed a fix to remove this unused import.

File: gemini-resin-archetype/src/main/resources/archetype-resources/src/main/java/com/techempower/handlers/HomeHandler.java (hunk 2 above)
Comment (k-r-g) [reply gemini-resin-archetype/src/main/resources/archetype-resources/src/main/java/com/techempower/handlers/HomeHandler.java]: My apologies for the commented-out bits of code. I just pushed a fix to remove those.

I switched to MethodUriHandler because it fits better with the server-side rendering we're demonstrating here. Happy to use MethodUriHandler instead if you can show me how it can show a server-side rendered 404 page? I'm probably missing something obvious.
